app/models/article.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Article < ApplicationRecord
+  has_many :articles_sales, dependent: :restrict_with_error
+  has_many :sales, through: :articles_sales
+  has_many :articles_refunds, dependent: :restrict_with_error
+  has_many :refunds, through: :articles_refunds
+
+  validates :name, presence: true, allow_blank: false
+  validates :price, presence: true, allow_blank: false,
+                    numericality: { greater_than_or_equal_to: 0, only_integer: true, message: 'Must be a positive
+                     number. Maximum 2 numbers after comma' }
```

Comment:
In the UI, the user will be ask to enter a decimal number and then it'll be multiply by 100

Reply:
Ah the price is in cents, I didn't catch that, makes sense! So we have two different validations:
- in the backend, we only expect integers
- in the frontend, we only expect float with a maximum of 2 decimals

In this case, I would change the error message in the backend to something like `Must be a positive integer` because it shouldn't be aware of how this integer is computed in the frontend. And have some logic in the frontend to display an error when the price is more than 2 decimals. What do you think?

Reply:
Yes I agree but I'm not sure on how to deal with that to change the error in front.